server/api/services/users.service.js

```diff
@@ -129,7 +130,8 @@ class UsersService {
         .find({ role: 'user', totalScore: { $gt: user.totalScore } })
         .countDocuments();
 
-      return { user, rank: rank + 1 };
+      const achievement = await achievementModel.find({ userId: roll }).populate('eventId');
```

Comment:
The event model has a lot of fields, populating the whole document would be expensive. Instead, we can just populate one field, i.e., the event name.